### DISCUSSION TITLE
How can I do Back Button work with custom keyboard?

### DISCUSSION THREAD

**Discussion by Sodikjon** [telegraf/telegraf#1593] [Q&A] [answered]
![Image-001-20-22-52_27-03-2022](https://user-images.githubusercontent.com/68941947/160288807-2c307f17-8cae-474b-bd89-6a8f9256fdb2.png)

Hi everybody!
How can I switch back to previous menu by clicking Back button?
See screenshot.
Thank you!

**Comment by MKRhere** [comment, marked as answer]
If you have only one main state (to which you can ever go "back" to), you can simply attach the same listener for `ctx.command("course", main)` and `ctx.hears("Back", main)`.

If you will have nested states ("Back" may lead to multiple previous menus), you may need to use Scenes. Consider checking out this post: https://dev.to/maklut/telegraf-vs-node-telegram-bot-api-36fk#telegraf-scenes
